src/platform/imx93_a55/lib/clk.c

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright 2023 NXP
+ */
+
+#include <rtos/clk.h>
+#include <sof/lib/notifier.h>
+
+static const struct freq_table platform_cpu_freq[] = {
+	{ CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC,
+		CONFIG_SYS_CLOCK_TICKS_PER_SEC * 1000 },
+};
+
+static struct clock_info platform_clocks_info[NUM_CLOCKS];
+
+void platform_clock_init(struct sof *sof)
+{
+	int i;
+
+	sof->clocks = platform_clocks_info;
+
+	/* The CCM doesn't seem to allow setting a core's
+	 * frequency. It probably sets the whole cluster's
+	 * frequency to some value (not relevant). Since
+	 * we're running on top of Jailhouse we don't want
+	 * to allow SOF to change the cluster's frequency
+	 * since that would also affect Linux.
```

Comment:
This is an interesting discussion to have -- how will Linux's own core frequency change affect Zephyr? Can it possibly mess with the playback in audible ways? Make it harder for the core to keep up if e.g. compress playback is happening?

Reply:
Very good question, thanks for posing it. I'm not sure how altering the cluster's frequency from Linux's side would affect SOF's flow as I haven't tested it out but this is definitely something we need to watch out for.